tlatools/org.lamport.tlatools/src/util/SimpleFilenameToStream.java

```diff
@@ -65,66 +70,53 @@ public SimpleFilenameToStream(final String libraryPath) {
    *
-   * @return the location of the TLA+ standard modules as an absolute path or as a "jar:" URL
+   * @return a resolver for locating the TLA+ standard modules
    */
-  private String getInstallationBasePath() {
-
-    // get a "file:" URL for the base directory for package tla2sany
-    final URL         url = cl.getResource(STANDARD_MODULES);
-
-    // jar expanded to the fs (make sure to handle whitespaces correctly
-    final String path = url.toString();
-	try {
-    	// convert to URI which handles paths correctly (even OS dependently)
-    	if(!FilenameToStream.isInJar(path)) {
-    	final URI uri = new URI(path);
-    		return new File(uri).getAbsolutePath();
-    	}
-    } catch (URISyntaxException e) {
-    	System.err.println(path);
-    	// may never happen
-    	e.printStackTrace();
-    } catch (IllegalArgumentException e) {
-    	System.err.println(path);
-    	// may never happen
-    	e.printStackTrace();
-    }
-    return path;
-   }
+  private static Resolver getInstallationBasePath() {
+    // First search the standard modules installation location, then search the
+    // bare classpath.  For historical reasons, modules discovered in both of these
+    // locations are considered "library" modules.
+    return new SequentialResolver(List.of(
+            new ClasspathResolver(cl, STANDARD_MODULES),
+            new ClasspathResolver(cl, "")));
+  }
 
   /**
    * August 2014 - TL
    * added an informative method for returning the actual path used by this resolver
    */
   public String getFullPath() {
-    StringBuffer buf = new StringBuffer();
-    String[] ar = libraryPaths;
-    for (int i=0; i<ar.length; i++)
-    {
-      buf.append(ar[i]);
-      if (i <ar.length-1) {
-        buf.append(", ");
-      }
-    }
-    return buf.toString();
+    return userFileResolver.describeSearchLocations() + ", " + standardLibraryResolver.describeSearchLocations();
   }
 
   /**
    * August 2014 - TL
    * added functionality for supplying additional libraries.
    * All usage of this except the one added by TL is supplying libraries  = null
    */
-  private String[] getLibraryPaths(final String installationBasePath, String[] libraries) {
-    String[] res;
-    String path = null;
+  private static Resolver getLibraryPaths(String[] libraries) {
+    List<Resolver> res = new ArrayList<>();
+
+    // First, if userDir is set, search there, otherwise search the current working directory.
+    String userDir = ToolIO.getUserDir();
+    if (userDir != null) {
+      res.add(new FilesystemResolver(Path.of(userDir)));
+    } else {
+      String cwd = System.getProperty("user.dir");
+      if (cwd != null) {
+        res.add(new FilesystemResolver(Path.of(cwd)));
+      }
+    }
+
+    // Next, search the given `libraries` (or the TLA_LIBRARY system property if `libraries` is null).
+    String path; // actually a list of paths separated by FileUtil.pathSeparator
     if (libraries == null) path = System.getProperty(TLA_LIBRARY);
```

Comment:
Perhaps you could also add `{…}` while you’re at it.

Reply:
I'll do that in a follow-up PR, since this method deserves some additional attention anyway (see [related comment](https://github.com/tlaplus/tlaplus/pull/1116#discussion_r1911629890)).